src/Store/CompletionStoreConnector.re

```diff
@@ -11,139 +11,192 @@ open Actions;
 
 module Option = Utility.Option;
 module VimEx = Utility.VimEx;
+module IndexEx = Utility.IndexEx;
 
 module Log = (val Log.withNamespace("Oni2.CompletionStore"));
 
-type lastCompletionMeet = {
-  completionMeetBufferId: int,
-  completionMeetLine: Index.t,
-  completionMeetColumn: Index.t,
-};
+module Effects = {
+  let requestCompletions =
+      (~languageFeatures, ~buffer, ~meet: CompletionMeet.t) =>
+    Isolinear.Effect.createWithDispatch(~name="requestCompletions", dispatch => {
+      Log.info("Requesting for " ++ Uri.toString(Buffer.getUri(buffer)));
 
-let defaultMeet = {
-  completionMeetBufferId: (-1),
-  completionMeetLine: Index.fromZeroBased(-1),
-  completionMeetColumn: Index.fromZeroBased(-1),
-};
+      let completionPromise =
+        LanguageFeatures.requestCompletions(~buffer, ~meet, languageFeatures);
 
-let lastMeet = ref(defaultMeet);
+      Lwt.on_success(completionPromise, completions => {
+        dispatch(CompletionAddItems(meet, completions))
+      });
+    });
 
-let lastBase = ref("");
+  let applyCompletion =
+      (~editor, ~meet: CompletionMeet.t, completion: CompletionItem.t) =>
+    Isolinear.Effect.createWithDispatch(~name="applyCompletion", dispatch => {
+      let cursor = Vim.Cursor.get();
+      let delta =
+        Index.toZeroBased(cursor.column)
+        - Index.toZeroBased(meet.location.column);
 
-let equals = (~line, ~column, ~bufferId, oldMeet) => {
-  oldMeet.completionMeetLine == line
-  && oldMeet.completionMeetColumn == column
-  && oldMeet.completionMeetBufferId == bufferId;
+      let _: list(Vim.Cursor.t) = VimEx.repeatInput(delta, "<BS>");
+      let cursors = VimEx.inputString(completion.label);
+
+      dispatch(EditorCursorMove(Editor.getId(editor), cursors));
+    });
 };
 
-let applyCompletion = (state: State.t) =>
-  Isolinear.Effect.createWithDispatch(~name="vim.applyCompletion", dispatch => {
-    let maybeFocused =
-      Option.map(
-        i => state.completions.filtered[i],
-        state.completions.focused,
+module Actions = {
```

Comment:
this module clashes a bit with `Core.Actions`, but this contains actual actions, while `Core.Actions.t` are just messages, which might describe actions, but ought to and now often does describe events instead. So this is another case where the the `Actions` name causes a bit of confusion.

Reply:
Sounds good. Per our discussion - we plan on moving `Core.Actions` -> `Core.Events`